developer/src/common/web/utils/src/types/kpj/kpj-file-reader.ts

```diff
@@ -13,20 +13,19 @@ export class KPJFileReader {
   public read(file: Uint8Array): KPJFile {
     let data: KPJFile;
 
-    const parser = new xml2js.Parser({
-      explicitArray: false,
-      mergeAttrs: false,
-      includeWhiteChars: false,
-      normalize: false,
-      emptyTag: ''
-    });
-
-    parser.parseString(file, (e: unknown, r: unknown) => {
-      if(e) {
-        throw e;
-      }
-      data = r as KPJFile;
-    });
+  const parser = new XMLParser({
+    ignoreAttributes: false, // We'd like attributes, please
+    attributeNamePrefix: '', // to avoid '@_' prefixes
+    numberParseOptions: {
+      skipLike: /(?:)/, // parse numbers as strings
+      hex: null,
+      leadingZeros: null,
+      eNotation: null,
+    },
+  });
+    const raw = parser.parse(file.toString());
+    delete raw['?xml']; // XML prologue
```

Comment:
I wonder if this is going to be the same across all our parsers, and if we should have a wrapper?

Reply:
maybe, though there are some nuances such as for the ldml test data. So I'm not sure it needs to be a wrapper.  Though a utility function to knock out the prologue might make sense.

Reply:
yup, we can let the dust settle first